cylc/flow/config.py

```diff
@@ -666,6 +652,25 @@ def __init__(
 
         self.mem_log("config.py: end init config")
 
+    def prelim_process_graph(self) -> None:
+        """Ensure graph is not empty; set integer cycling mode and icp/fcp = 1
+        for simplest "R1 = foo" type graphs.
+        """
+        graphdict = self.cfg['scheduling']['graph']
+        if not any(graphdict.values()):
+            raise SuiteConfigError('No suite dependency graph defined.')
+
+        if (
+            'cycling mode' not in self.cfg['scheduling'] and
+            self.cfg['scheduling'].get('initial cycle point', '1') == '1' and
+            all(item in ['graph', '1', 'R1'] for item in graphdict)
```

Comment:
This doesn't pick up on equivalent recurrences to `R1`, e.g. `R1/1/1`, but that can come later